spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDeploymentController.java

```diff
@@ -63,11 +65,13 @@ public TaskDeploymentController(TaskService taskService) {
 	 * @param taskName the name of the existing task to be executed (required)
 	 * @param properties the runtime properties for the task, as a comma-delimited list of
 	 * 					 key=value pairs
+	 * @param params the runtime commandline arguments
 	 */
 	@RequestMapping(value = "/{name}", method = RequestMethod.POST)
 	@ResponseStatus(HttpStatus.CREATED)
-	public void deploy(@PathVariable("name") String taskName, @RequestParam(required = false) String properties) {
-		this.taskService.executeTask(taskName, DeploymentPropertiesUtils.parse(properties));
+	public void deploy(@PathVariable("name") String taskName, @RequestParam(required = false) String properties,
+			@RequestParam(required = false) List<String> params) {
+		this.taskService.executeTask(taskName, DeploymentPropertiesUtils.parse(properties), params);
```

Comment:
Lets take a default `format` property and pass it in via params as `--format="yyyy-MM-dd HH:mm:ss.SSS"`.

It has a space in it between `dd` and `HH`. We can't just start to split these with spaces because then one parameter would be splitted into to two parameters.

I'm now also seeing some weird behaviour with local deployer where multiple params gets mixed up somehow. Something to do with escaping or ways how stuff is started.